src/modules/launcher/Plugins/Microsoft.Plugin.Settings/Main.cs

```diff
@@ -0,0 +1,132 @@
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.Plugin.Settings.Enumerations;
+using Microsoft.Plugin.Settings.Helper;
+using Microsoft.Win32;
+using Wox.Plugin;
+
+namespace Microsoft.Plugin.Settings
+{
+    public sealed class Main : IPlugin
+    {
+        private IEnumerable<IWindowsSetting>? _settingsList;
+
+        public void Init(PluginInitContext context)
+        {
+            var currentWindowsVersion = ushort.MinValue;
+
+            try
+            {
+                var releaseIdString = Registry.GetValue(
+                    "HKEY_LOCAL_MACHINE\\SOFTWARE\\WOW6432Node\\Microsoft\\Windows NT\\CurrentVersion",
+                    "ReleaseId",
+                    null) as string;
+
+                if (!ushort.TryParse(releaseIdString, out currentWindowsVersion))
+                {
+                    currentWindowsVersion = ushort.MinValue;
+                }
+            }
+            catch (Exception exception)
+            {
+                Debug.WriteLine(exception);
+
+                // fall-back
+                currentWindowsVersion = ushort.MinValue;
+            }
+
+            _settingsList = SettingsHelper.GetSettings();
+
+            // remove deprecated settings Windows settings
+            _settingsList = _settingsList
+                .Where(found => currentWindowsVersion < found.DeprecatedInVersion);
+
+            // remove settings that are for a higher Windows versions
+            _settingsList = _settingsList
+                .Where(found => currentWindowsVersion >= found.IntroducedInVersion);
+
+            // sort settings list
+            _settingsList = _settingsList
+                .OrderBy(found => found.Name)
+                .ThenBy(found => found.Link);
+        }
+
+        public List<Result> Query(Query query)
+        {
+            var filteredList = _settingsList.Where(found
+                => found.Link.Contains(query.Search, StringComparison.InvariantCultureIgnoreCase)
+                || found.Name.Contains(query.Search, StringComparison.InvariantCultureIgnoreCase)
+                || found.Area.ToString().Contains(query.Search, StringComparison.InvariantCultureIgnoreCase));
+
+            var newList = new List<Result>(filteredList.Count());
+
+            foreach (var entry in filteredList)
+            {
+                if (entry.Area == Area.ControlPanel)
+                {
+                    newList.Add(GetControlPanelResult(entry));
+                }
+                else
+                {
+                    newList.Add(GetWindowsSettingsResult(entry));
+                }
+            }
+
+            return newList;
+        }
+
+        private Result GetWindowsSettingsResult(IWindowsSetting entry)
+        {
+            var subTitle = string.IsNullOrEmpty(entry.Note)
+                    ? $"Ares: {SettingsHelper.GetAreaText(entry)}"
```

Comment:
what is ares?
same below

Reply:
Its the settings area, taken from the documentation site
https://docs.microsoft.com/en-us/windows/uwp/launch-resume/launch-settings-app#ms-settings-uri-scheme-reference
should i rename it to "Category" ?